app/scripts/controllers/editRegistrationModal.js

```diff
@@ -12,45 +13,37 @@ angular.module('confRegistrationWebApp')
       $modalInstance.dismiss();
     };
 
-
-    //remove blocks that are not part of registrant type
-    angular.forEach(angular.copy(conference.registrationPages), function(page) {
-      var pageIndex = _.findIndex($scope.conference.registrationPages, { 'id': page.id });
-      angular.forEach(angular.copy(page.blocks), function(block) {
-        //check if block has been removed for this registrant type
-        //check if block type is paragraphContent
-        if (_.contains(block.registrantTypes, registrant.registrantTypeId) || block.type === 'paragraphContent') {
-          _.remove($scope.conference.registrationPages[pageIndex].blocks, function(b) { return b.id === block.id; });
-        }
-      });
-
-      if($scope.conference.registrationPages[pageIndex].blocks.length === 0) {
-        _.remove($scope.conference.registrationPages, function(p) { return p.id === page.id; });
-      }
-    });
-
+    $scope.blockInRegType = function(block, regTypeId){
+      return !_.contains(block.registrantTypes, regTypeId);
+    };
 
     $scope.submit = function (setRegistrationAsCompleted) {
       $scope.saving = true;
 
-      angular.forEach($scope.adminEditRegistration.answers, function(a){
-        if(!angular.equals(a, _.find(registrant.answers, { 'id': a.id }))){
+      angular.forEach($scope.adminEditRegistrant.answers, function(a){
+        if(!angular.equals(a, _.find(originalRegistrantObject.answers, { 'id': a.id }))){
           answersToUpdate.push(a);
         }
       });
 
       if(setRegistrationAsCompleted){
-        var r = confirm('Are you sure you want to mark this registration as completed?');
-        if (!r) {
+        if (!confirm('Are you sure you want to mark this registration as completed?')) {
           $scope.saving = false;
           return;
         }
         $scope.registration.completed = true;
-        $http.put('registrations/' + registrant.registrationId, $scope.registration).success(function(){
+        $http.put('registrations/' + originalRegistrantObject.registrationId, $scope.registration).success(function(){
           saveAnswer();
```

Comment:
Why does this call `saveAnswer()` and put all the changed blocks after having put the entire registrant on the line before it? I seems line the `$http.put()` already saved the new answers to the server. Same goes for line 42